### ISSUE TITLE
Cyclic references when rendering enum constant 

### ISSUE THREAD

**Issue by oowekyala** [snowe2010/pretty-print#19]
First, thanks for this library! 

Enum constants are weirdly rendered though. Say I have 
```kotlin
enum class ScalarType {
    NUMBER,
    STRING,
    BOOLEAN,
    NULL;

    override fun toString(): String = name
}
```
Then for some reason,
```kotlin
            ScalarType.BOOLEAN.pp()
```
renders as 
```
ScalarType(
  NUMBER = ScalarType(
    NUMBER = cyclic reference detected for 1183231938
    STRING = ScalarType(
      NUMBER = cyclic reference detected for 1183231938
      STRING = cyclic reference detected for 912672706
      BOOLEAN = cyclic reference detected for 504053874
      NULL = ScalarType(
        NUMBER = cyclic reference detected for 1183231938
        STRING = cyclic reference detected for 912672706
        BOOLEAN = cyclic reference detected for 504053874
        NULL = cyclic reference detected for 2032079962
      )[$id=2032079962]
    )[$id=912672706]
    BOOLEAN = cyclic reference detected for 504053874
    NULL = ScalarType(
      NUMBER = cyclic reference detected for 1183231938
      STRING = ScalarType(
        NUMBER = cyclic reference detected for 1183231938
        STRING = cyclic reference detected for 912672706
        BOOLEAN = cyclic reference detected for 504053874
        NULL = cyclic reference detected for 2032079962
      )[$id=912672706]
      BOOLEAN = cyclic reference detected for 504053874
      NULL = cyclic reference detected for 2032079962
    )[$id=2032079962]
  )[$id=1183231938]
  STRING = ScalarType(
    NUMBER = ScalarType(
      NUMBER = cyclic reference detected for 1183231938
      STRING = cyclic reference detected for 912672706
      BOOLEAN = cyclic reference detected for 504053874
      NULL = ScalarType(
        NUMBER = cyclic reference detected for 1183231938
        STRING = cyclic reference detected for 912672706
        BOOLEAN = cyclic reference detected for 504053874
        NULL = cyclic reference detected for 2032079962
      )[$id=2032079962]
    )[$id=1183231938]
    STRING = cyclic reference detected for 912672706
    BOOLEAN = cyclic reference detected for 504053874
    NULL = ScalarType(
      NUMBER = ScalarType(
        NUMBER = cyclic reference detected for 1183231938
        STRING = cyclic reference detected for 912672706
        BOOLEAN = cyclic reference detected for 504053874
        NULL = cyclic reference detected for 2032079962
      )[$id=1183231938]
      STRING = cyclic reference detected for 912672706
      BOOLEAN = cyclic reference detected for 504053874
      NULL = cyclic reference detected for 2032079962
    )[$id=2032079962]
  )[$id=912672706]
  BOOLEAN = cyclic reference detected for 504053874
  NULL = ScalarType(
    NUMBER = ScalarType(
      NUMBER = cyclic reference detected for 1183231938
      STRING = ScalarType(
        NUMBER = cyclic reference detected for 1183231938
        STRING = cyclic reference detected for 912672706
        BOOLEAN = cyclic reference detected for 504053874
        NULL = cyclic reference detected for 2032079962
      )[$id=912672706]
      BOOLEAN = cyclic reference detected for 504053874
      NULL = cyclic reference detected for 2032079962
    )[$id=1183231938]
    STRING = ScalarType(
      NUMBER = ScalarType(
        NUMBER = cyclic reference detected for 1183231938
        STRING = cyclic reference detected for 912672706
        BOOLEAN = cyclic reference detected for 504053874
        NULL = cyclic reference detected for 2032079962
      )[$id=1183231938]
      STRING = cyclic reference detected for 912672706
      BOOLEAN = cyclic reference detected for 504053874
      NULL = cyclic reference detected for 2032079962
    )[$id=912672706]
    BOOLEAN = cyclic reference detected for 504053874
    NULL = cyclic reference detected for 2032079962
  )[$id=2032079962]
)[$id=504053874]
```

**Comment by snowe2010** [comment]
Fantastic! I've found the issue, but first I'd like to ask what you would desire the output to be? I was thinking either 

```
BOOLEAN
```

or 

```
ScalarType.BOOLEAN
```

**Comment by oowekyala** [comment]
Nice work! 

I don't mind either way. I think `ScalarType.BOOLEAN` is maybe clearer, especially since if it is eg a field of a data class, it will be rendered on its own line so that the `ScalarType.` prefix isn't really wasting space. That's how I use the enum at least, I don't know if in other contexts the prefix would be noisy.

**Comment by snowe2010** [comment]
Closed in v2.0.2. (I think something might have gone wrong with the deploy though, so if you see anything wrong please let me know!)
